## TensorFlow

In [ ]:
import tensorflow as tf
print("TensorFlow version:", tf.__version__)

TensorFlow version: 2.15.0


In [ ]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

In [ ]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])

In [ ]:
predictions = model(x_train[:1]).numpy()
predictions

array([[ 0.6489673 ,  0.2793069 , -0.92776126, -0.29420823, -0.4965938 ,
         0.11540303, -0.31736195,  1.2832264 , -0.4046147 ,  0.13251327]],
      dtype=float32)

In [ ]:
tf.nn.softmax(predictions).numpy()

array([[0.15617715, 0.10791356, 0.03227402, 0.06081372, 0.04967143,
        0.09159959, 0.05942183, 0.29449147, 0.05445687, 0.09318035]],
      dtype=float32)

In [ ]:
loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

In [ ]:
loss_fn(y_train[:1], predictions).numpy()

2.3903286

In [ ]:
model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

In [ ]:
model.fit(x_train, y_train, epochs=5)

Epoch 1/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.2898 - accuracy: 0.9163
Epoch 2/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.1413 - accuracy: 0.9584
Epoch 3/5
1875/1875 [==============================] - 6s 3ms/step - loss: 0.1058 - accuracy: 0.9678
Epoch 4/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.0859 - accuracy: 0.9738
Epoch 5/5
1875/1875 [==============================] - 8s 4ms/step - loss: 0.0723 - accuracy: 0.9771


In [ ]:
model.evaluate(x_test,  y_test, verbose=2)

313/313 - 1s - loss: 0.0722 - accuracy: 0.9783 - 1s/epoch - 5ms/step


[0.07223807275295258, 0.9782999753952026]

## The image classifier is now trained to ~98% accuracy on this dataset.

ABOVE 👆🏻👆🏻👆🏻